src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -24,7 +24,7 @@ public static bool IsInvalid(this IOperation operation, Compilation compilation,
 
             // if wrong compilation is given, GetSemanticModel will throw due to tree not belong to the given compilation.
             var model = compilation.GetSemanticModel(operation.Syntax.SyntaxTree);
-            return model.GetDiagnostics(operation.Syntax.Span, cancellationToken).Any(d => d.Severity == DiagnosticSeverity.Error);
+            return model.GetDiagnostics(operation.Syntax.Span, cancellationToken).Any(d => d.DefaultSeverity == DiagnosticSeverity.Error);
```

Comment:
sure I can change it to HasErrors.

...

my understanding of this API is that, we want to provide an API to skip analysis if something in code is not right for analysis. all of usage cases I have seen so far either in our code (roslyn analyzers or test analyzers) or customers code are calling IsInvalid right away in registered action and bail out right away if operation is *Invalid*.

are we trying to change this usage pattern? or are we just trying to provide different API for same usage pattern?

it sounds like people are arguing about implementation not what user is expected to use this API.

do we want people to use this API (HasErrors, HasDiagnostics or IsInvalid) other than the one we have seen so far?